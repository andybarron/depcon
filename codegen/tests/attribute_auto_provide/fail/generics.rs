use depcon::*;

trait Interface<A> {}

#[derive(Injectable)]
struct Implementation;

#[auto_provide]
impl<A> Interface<A> for Implementation {}

fn main() {}
