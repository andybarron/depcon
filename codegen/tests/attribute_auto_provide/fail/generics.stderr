error: custom attribute panicked
 --> tests/attribute_auto_provide/fail/generics.rs:8:1
  |
8 | #[auto_provide]
  | ^^^^^^^^^^^^^^^
  |
  = help: message: Only concrete types can be registered. Use auto_register! instead
